src/Bicep.Core/Bicep.Core.csproj

```diff
@@ -5,6 +5,9 @@
     <EnableNuget>true</EnableNuget>
     <PackageTags>Azure;ResourceManager;ARM;Deployments;Templates;Bicep</PackageTags>
     <Description>Bicep compiler core functionality</Description>
+
+    <!-- enable trimmer warnings for Bicep CLI and projects it references -->
```

Comment:
Worth just enabling this at the solution level? Even for test projects I can see an argument for trimming - so that we're more likely to catch potential issues with the CLI.

Reply:
Trimming imposes some fairly strict requirements on your entire transitive closure. If all your dependencies are trimmable (no reflection, no Newtonsoft, using source generators with STJ, etc.) it works great and is very clean. If you don't meet those requirements (Bicep CLI does not), it becomes rather developer-hostile. There's no help with the suppressions and the full set of trimming errors are only available only at `dotnet publish` time. (Analyzers I enabled only give partial diagnostics.)

 We should not be using it anywhere else. The ROI is just not there.

I am thinking of introducing an import for shared trimming settings though to make what we do trim more maintainable.